aten/src/ATen/native/Histogram.cpp

```diff
@@ -0,0 +1,133 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+
+#include <ATen/native/Histogram.h>
+
+#include <tuple>
+
+namespace at { namespace native {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_stub);
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_linear_stub);
+
+namespace {
+
+inline void histogram_pre_check(const Tensor& input, const Tensor& bins, const c10::optional<Tensor>& weight,
+    bool density, const Tensor& hist, const Tensor& bin_edges) {
+  TORCH_CHECK(bins.device() == input.device(), "bins and input value tensors should have same device type, ",
+    "but we got bins tensor device type ", bins.device(), " and input value tensor device type ", input.device());
+
+  TORCH_CHECK(bins.dim() == 1, "bins tensor should have dimension 1, but got ", bins.dim(), " dimension");
+
+  TORCH_CHECK(hist.dim() == 1, "hist tensor should have dimension 1, but got ", hist.dim(), " dimension");
+
+  TORCH_CHECK(bin_edges.dim() == 1, "bin_edges tensor should have dimension 1, but got ", bin_edges.dim(), " dimension");
```

Comment:
Because `bin_edges` is an out= tensor we should use `resize_output` and let that determine if an error should be thrown. 

See https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch for more information